lib/oli_web/live/collaboration_live/collab_space_view.ex

```diff
@@ -266,18 +249,16 @@ defmodule OliWeb.CollaborationLive.CollabSpaceView do
   end
 
   # -----------Edit Post-----------#
+  def handle_event("set_editing_post", %{"post_id" => post_id}, socket) do
+    editing_post = socket.assigns.editing_post
 
-  def handle_event("display_edit_modal", %{"id" => id}, socket) do
-    post = Collaboration.get_post_by(%{id: id})
-    changeset = Collaboration.change_post(post)
-
-    modal_assigns = %{
-      id: "edit_post_modal",
-      on_submit: "edit_post",
-      changeset: changeset
-    }
+    {is_edition_mode, post} =
+      case editing_post && Integer.to_string(editing_post.id) == post_id do
+        true -> {false, nil}
+        _ -> {true, Collaboration.get_post_by(%{id: post_id})}
+      end
 
-    display_post_modal(modal_assigns, assign(socket, editing_post: post))
```

Comment:
If we are not going to use the modal, then we can remove the `OliWeb.CollaborationLive.Posts.Modal` alias